File: src/main/java/com/backend/imechanic/model/Vehicle.java
package com.backend.imechanic.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "vehicles")
@Builder
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String plate;

    @Column(nullable = false)
    private String model;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String year;

    @Column(nullable = false)
    private boolean active;

    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private UserEntity customer;

    @OneToMany(mappedBy = "vehicle")
    private List<Order> orders;

    @Column(name = "created_at")
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: frontend/src/workshop/models/vehicle.ts
export interface Vehicle {
  id: string;
  plate: string;
  brand: string;
  model: string;
  year: number;
  firstName: string;
  lastName: string;
}

export type VehicleCreate = {
    plate: string;
    model: string;
    brand: string;
    year: string;
    firstName: string;
    lastName: string;
    phone: string;
}

export type VehicleCreateResponse = {
    plate: string;
    model: string;
    brand: string;
    year: string;
    firstName: string;
    lastName: string;
    phone: string;
}